### PR TITLE
CI: reduce the test matrix

### PR DESCRIPTION
**Pull request by AkihiroSuda**
For the sake or reducing test flakes

### CODE REVIEW

File: .github/workflows/lima.yaml [hunk 1]
```diff
@@ -13,17 +13,7 @@ permissions:
...
-      matrix:
-        # macos-13-large is used as macos-13 seems too flaky.
-        # macos-14 (ARM) and later cannot be used for the most part of the job
-        # due to the lack of the support for nested virt.
-        platform:
-          - macos-13-large
-          - macos-14-large
-          - macos-15-large
-    runs-on: ${{ matrix.platform }}
+    runs-on: macos-15-large
```
Comment (nirs) [inline .github/workflows/lima.yaml:16]: I would keep `{{ matrix.platform }}` and a list with one item so adding more items is easier later. This also generates more useful name for the job (integration (macos-15)).

File: .github/workflows/lima.yaml [hunk 2]
```diff
@@ -53,8 +43,7 @@ jobs:
           make ARCH=arm64
           sudo make PREFIX=/opt/socket_vmnet.arm64 install.bin install.doc
           if file /opt/socket_vmnet.arm64/bin/* | grep -q x86_64 ; then false ; fi
-      - name: Update brew (macos-15-large)
-        if: matrix.platform == 'macos-15-large'
+      - name: Update brew
```
Comment (nirs) [inline .github/workflows/lima.yaml:46]: I would keep this as well, it documents why the update is needed. If we add another platform later the unneeded step will be skippped.

File: .github/workflows/lima.yaml (hunk 1 above)
Comment (AkihiroSuda) [reply .github/workflows/lima.yaml]: fixed

File: .github/workflows/lima.yaml (hunk 2 above)
Comment (AkihiroSuda) [reply .github/workflows/lima.yaml]: fixed

**Review by nirs — APPROVED** [review]

**Pull request merged**